### PR TITLE
move build completion report to subaction

### PR DESCRIPTION
**Pull request by ewlu**

### CODE REVIEW

File: .github/actions/report-build-complete-status/action.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+              repo: context.repo.repo,
+              labels: ['build-failure']
+            })
+
+      - name: Report error on build-failure
+        if: ${{ always() && inputs.build_status == 'failure' }}
+        shell: bash
+        working-directory: riscv-gnu-toolchain
+        run: |
+          python scripts/post_check_to_patchworks.py -event ${{ github.event_name }} -repo ewlu/gcc-precommit-ci -pid $(tail -n 1 patches/${{ inputs.patch_name }})  -desc 'Build failed' -iid '${{ inputs.issue_num }}#issuecomment-${{ inputs.build_comment_id }}' -state 'fail' -context 'build--${{ inputs.target }}' -token '${{ secrets.PATCHWORK_API }}'
```
Comment (patrick-rivos) [inline .github/actions/report-build-complete-status/action.yaml:83]: Use input patchwork_api

File: .github/actions/report-build-complete-status/action.yaml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        continue-on-error: true
+
+      - name: Report build success
+        shell: bash
+        working-directory: riscv-gnu-toolchain
+        run: |
+          python scripts/post_check_to_patchworks.py -event ${{ github.event_name }} -repo ewlu/gcc-precommit-ci -pid $(tail -n 1 patches/${{ inputs.patch_name }})  -desc 'Build passed' -iid '${{ inputs.issue_num }}#issuecomment-${{ inputs.build_comment_id }}' -state 'success' -context 'build--${{ inputs.target }}' -token '${{ secrets.PATCHWORK_API }}'
```
Comment (patrick-rivos) [inline .github/actions/report-build-complete-status/action.yaml:90]: Ditto

**Review by patrick-rivos — CHANGES_REQUESTED** [review]
Overall looks good to me. That's a _lot_ of inputs but keeping these two text-dense step series in sync is a good idea.

**Review by patrick-rivos — APPROVED** [review]
LGTM - Just a heads up/highlighting that this will cause patchworks build reporting to be on as soon as this is merged.

**Comment by ewlu** [comment]
It won't be sending build reporting due to https://github.com/ewlu/gcc-precommit-ci/commit/4ae6f86c3e5094bc983878d0b92679fcab2ac3d6 at the start of the refactor branch

**Pull request merged**